source/common/ssl/context_manager_impl.cc

```diff
@@ -22,6 +22,10 @@ void ContextManagerImpl::releaseContext(Context* context) {
 
 ClientContextPtr ContextManagerImpl::createSslClientContext(Stats::Scope& scope,
                                                             const ClientContextConfig& config) {
+  if (!config.isValid()) {
```

Comment:
How is this going to work on the future? Can you take me through the logic that is coming in the follow up update PRs so I understand? Right now, this will return a null context which is fine, things will fail (and we can have integration tests in this PR just proving things fail). But in the future, how is the context going to get updated inside the transport factory?

Reply:
I copied this from my other comment.

In the next PR, SocketFactory will register a UpdateCallback to SDS_api, if SDS secret is fetched or updated,  it will call the SocketFactory::onUpdate(), SocketFactory will call ContextManagerImpl::createClientContext()  to create a valid Context if ContextConfig is valid. This is called in the main thread too since SDS_api::onConfigUpdate will be called in the main thread.